src/PortalRegistry.sol

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.21;
+
+import { Initializable } from "openzeppelin-contracts/contracts/proxy/utils/Initializable.sol";
+import { ERC165Checker } from "openzeppelin-contracts/contracts/utils/introspection/ERC165Checker.sol";
+import { IPortal } from "./interface/IPortal.sol";
+import { Portal } from "./struct/Portal.sol";
+
+/**
+ * @title Portal Registry
+ * @author Consensys
+ * @notice This contract aims to manage the Portals used by attestation issuers
+ */
+contract PortalRegistry is Initializable {
+  mapping(address id => Portal portal) private portals;
+  address[] private portalAddresses;
+
+  /// @notice Error thown when attempting to register a Portal twice
+  error PortalAlreadyExists();
+  /// @notice Error thrown when attempting to register a Portal that is not a smart contract
+  error PortalAddressInvalid();
+  /// @notice Error thown when attempting to register a Portal with an empty name
+  error PortalNameMissing();
+  /// @notice Error thrown when attempting to register a Portal with an empty description
+  error PortalDescriptionMissing();
+  /// @notice Error thrown when attempting to register a Portal that does not implement IPortal interface
+  error PortalInvalid();
+  /// @notice Error thrown when attempting to get a Portal that is not registered
+  error PortalNotRegistered();
+
+  /// @notice Event emitted when a Portal registered
+  event PortalRegistered(string name, string description, address moduleAddress);
+
+  /**
+   * @notice Contract initialization
+   */
+  function initialize() public initializer {}
+
+  /**
+   * @notice Registers a Portal to the PortalRegistry
+   * @param id the portal name
```

Comment:
Are these code comments correct? Should `id` actually be described as "portal contract address and unique identifier" or something